src/fast-canon-map.c

```diff
@@ -37,46 +40,44 @@ void decimalToBitArray(int bitMatrix[k][k], unsigned long D){
 unsigned long bitArrayToDecimal(int bitMatrix[k][k], char Permutations[], int numBits){
     unsigned long num=0;
     int lf=0;
-    for(int i = 0; i < k-1; i++)
-	for(int j=i+1; j < k; j++){
-	    num+=(((unsigned long)bitMatrix[(int)Permutations[i]][(int)Permutations[j]]) << (numBits-1-lf));
-	    lf++;
-	}
+    for(int i = 0; i < k; i++)
+    for(int j = (i+1)*(1-directed); j < k; j++){
+	if(i==j) continue;
+	num+=(((unsigned long)bitMatrix[(int)Permutations[i]][(int)Permutations[j]]) << (numBits-1-lf)); 
+	lf++;
+    }
     return num;
 }
 
 void decimalToBitArray(int bitMatrix[k][k], unsigned long D){
-    for(int i=k-2; i>=0; i--)
-	for(int j=k-1; j>i; j--){
+    for(int i=k-1; i>=0; i--)
+	for(int j=k-1; j>=(i+1)*(1-directed); j--){
+	    if(i==j) continue;
 	    bitMatrix[i][j] = D%2;
-	    bitMatrix[j][i]=bitMatrix[i][j];
 	    D = D/2;
 	}
 }
 
 #endif
 
 
-
 typedef unsigned char xChar[5];//40 bits for saving index of canonical decimal and permutation
 
 static xChar* data;
 static bool* done;
-static unsigned long canonicalDecimal[12346];//12346 canonical graphettes for k=8
+static unsigned long canonicalDecimal[1600000];
```

Comment:
Please give this constant a name and a comment (I know, I didn't do it for 12346, but "do as I say, not as I do" :-). Also make the constant exact rather than 16 with a bunch of zeros afterwards.